pipeline_templates/codeql3000_default.yml

```diff
@@ -0,0 +1,80 @@
+# This is the job template for running CodeQL on a repository
+
+parameters:
+- name: 'cmake_command'
+  type: string
+  default: '.. -G "Visual Studio 17 2022" -A x64'
+
+jobs:
+- job: codeql
+  displayName: 'Run CodeQL on production code'
+  pool:
+    name: Azure-MessagingStore-WinBuildPoolVS2022
+    demands:
+    - Cmd
+    - msbuild
+    - visualstudio
+    - build
+
+  workspace:
+    clean: all
+
+  steps:
+  - task: BatchScript@1
+    displayName: 'Git submodule update'
+    inputs:
+      filename: 'C:\Program Files\Git\bin\git.exe'
+      arguments: 'submodule update --init --force'
+
+  - task: BatchScript@1
+    displayName: 'Git submodule clean'
+    inputs:
+      filename: 'C:\Program Files\Git\bin\git.exe'
+      arguments: 'submodule foreach --recursive "git clean -xdff"'
+
+  - task: BatchScript@1
+    displayName: 'Git clean'
+    inputs:
+      filename: 'C:\Program Files\Git\bin\git.exe'
+      arguments: 'clean -xdff'
+
+  - task: DeleteFiles@1
+    displayName: 'Clean build binaries directory'
+    inputs:
+      sourceFolder: $(Build.BinariesDirectory)
+      contents: \*
+
+  - task: BatchScript@1
+    displayName: 'Setup VS Vars'
+    inputs:
+      filename: '"c:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\VsDevCmd.bat"'
+      modifyEnvironment: true
+
+  - template: codeql3000_init.yml
+
+  - task: CMake@1
+    displayName: 'CMake ${{parameters.cmake_command}}'
+    inputs:
+      workingDirectory: 'build_x64'
```

Comment:
>build_x64 [](http://example.com/codeflow?start=25&length=9)

If we are thoroughly cleaning git/git submodules/BuildBinariesDirectory we should clean the cmake folder as well while we are here.

Reply:
I think we do by the virtue that we build usually in the source tree (in build_x64 or similar).